src/cunumeric/fft/fft.cu

```diff
@@ -0,0 +1,392 @@
+/* Copyright 2021 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include <csignal>
+
+#include "cunumeric/fft/fft.h"
+#include "cunumeric/fft/fft_template.inl"
+
+#include "cunumeric/cuda_help.h"
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+using dim_t = long long int;
+
+template <int DIM>
+__host__ static inline void cufft_operation(void* output,
+                                            void* input,
+                                            const Rect<DIM>& out_rect,
+                                            const Rect<DIM>& in_rect,
+                                            std::vector<int64_t>& axes,
+                                            fftType type,
+                                            fftDirection direction)
+{
+  auto stream = get_cached_stream();
+
+  size_t workarea_size = 0;
+  size_t num_elements;
+  dim_t n[DIM];
+  dim_t inembed[DIM];
+  dim_t onembed[DIM];
+
+  const Point<DIM> zero   = Point<DIM>::ZEROES();
+  const Point<DIM> one    = Point<DIM>::ONES();
+  Point<DIM> fft_size_in  = in_rect.hi - in_rect.lo + one;
+  Point<DIM> fft_size_out = out_rect.hi - out_rect.lo + one;
+  num_elements            = 1;
+  for (int i = 0; i < DIM; ++i) {
+    n[i] =
+      (type == fftType::FFT_R2C || type == fftType::FFT_D2Z) ? fft_size_in[i] : fft_size_out[i];
+    inembed[i] = fft_size_in[i];
+    onembed[i] = fft_size_out[i];
+    num_elements *= n[i];
+  }
+
+  // Create the plan
+  cufftHandle plan;
+  CHECK_CUFFT(cufftCreate(&plan));
```

Comment:
Any reason that you didn't use the plan cache? If the cache doesn't give you what you want, you could either extend it your self or ask me to do it.